sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/SchemaRegistryAvroDeserializerBuilder.java

```diff
@@ -0,0 +1,90 @@
+
+/**
+ * Builder class for constructing {@link SchemaRegistryAvroDeserializer} and {@link SchemaRegistryAvroAsyncDeserializer}
+ */
+public class SchemaRegistryAvroDeserializerBuilder {
+
+    private final String registryUrl;
+    private TokenCredential credential;
+    private boolean avroSpecificReader;
+    private Integer maxSchemaMapSize;
+
+    /**
+     * Instantiates instance of Builder class.
+     * Supplies client defaults.
+     *
+     * @param registryUrl base schema registry URL for storing and fetching schemas
+     */
+    public SchemaRegistryAvroDeserializerBuilder(String registryUrl) {
+        this.registryUrl = registryUrl;
+        this.credential = null;
+        this.avroSpecificReader = false;
+        this.maxSchemaMapSize = null;
+    }
+
+    /**
+     *
+     * @param credential TokenCredential to be used for authenticating with Azure Schema Registry Service
+     * @return updated {@link SchemaRegistryAvroDeserializerBuilder} instance
+     */
+    public SchemaRegistryAvroDeserializerBuilder credential(TokenCredential credential) {
+        this.credential = credential;
+        return this;
+    }
+
+    /**
+     * Specifies if objects should be deserialized into Avro SpecificRecords via Avro SpecificDatumReader
+     * @param avroSpecificReader specific reader flag
+     * @return updated {@link SchemaRegistryAvroDeserializerBuilder} instance
+     */
+    public SchemaRegistryAvroDeserializerBuilder avroSpecificReader(boolean avroSpecificReader) {
+        this.avroSpecificReader = avroSpecificReader;
+        return this;
+    }
+
+    /**
+     * Specifies maximum schema object cache size for underlying CachedSchemaRegistryClient.  If specified cache
+     * size is exceeded, all caches are recycled.
+     *
+     * @param maxSchemaMapSize maximum number of schemas per cache
+     * @return updated {@link SchemaRegistryAvroDeserializerBuilder} instance
+     */
+    public SchemaRegistryAvroDeserializerBuilder maxSchemaMapSize(int maxSchemaMapSize) {
+        this.maxSchemaMapSize = maxSchemaMapSize;
+        return this;
+    }
+
+    /**
+     * Construct instance of {@link SchemaRegistryAvroAsyncDeserializer}
+     *
+     * @return {@link SchemaRegistryAvroAsyncDeserializer} instance
+     *
+     * @throws NullPointerException if parameters are incorrectly set.
+     * @throws IllegalArgumentException if credential is not set.
+     */
+    public SchemaRegistryAvroAsyncDeserializer buildAsyncClient() {
+        return new SchemaRegistryAvroAsyncDeserializer(this.buildSyncClient());
+    }
+
+    /**
+     * Construct instance of {@link SchemaRegistryAvroDeserializer}
+     *
+     * @return {@link SchemaRegistryAvroDeserializer} instance
+     *
+     * @throws NullPointerException if parameters are incorrectly set.
+     * @throws IllegalArgumentException if credential is not set.
+     */
+    public SchemaRegistryAvroDeserializer buildSyncClient() {
```

Comment:
Check if `credential` and `registryUrl` are null before creating the `Deserializer`.

Reply:
I am going update this such that the CachedSchemaRegistryClient is created in the builder and is passed into the constructor of the serializer/deserializer.

Takes advantage of all the checks in CachedSchemaRegistryClientBuilder.